adam.c

```diff
@@ -0,0 +1,162 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   adam.c                                             :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: nkhribec <marvin@42.fr>                    +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2019/09/22 15:32:50 by nkhribec          #+#    #+#             */
+/*   Updated: 2019/10/05 01:54:59 by nkhribec         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "header.h"
+
+void	printb(unsigned int c)
+{
+	int             i;
+	unsigned int	m;
+
+	i = 32;
+	while (--i > -1)
+	{
+		m = (((c & (1 << i)) >> i) + '0');
+		write(1, &m, 1);
+	}
+	write(1, "\n", 1);
+}
+
+int		ft_putunbr(unsigned long long n, int flag, int precision, int width)//also handle h, hh, l, ll
```

Comment:
you shouldn't print the PLUS in the case of unsigned numbers (%u %x %o), all of those are printed without + even if the flag is activated.

Reply:
you're right, i had a bad understanding to this, thanks.